ranger/ranger.go

```diff
@@ -44,10 +47,49 @@ type Ranger struct {
 	users     middleware.UserStorer
 }
 
+/*
+type Config struct {
+	UserStore func(db postgres.DatabaseService) middleware.UserStorer
+}
+*/
```

Comment:
This was another option. Instead of the `Config` itself having the type parameter, below, the commented out `func UserStorer` does. I opted for `Config` itself since there's no chance for misuse: you can't init a `Config` with instantiating it with a type. With this alternative, one could always leave `UserStore` `nil` unintentionally.

Reply:
I'm unsure here. The win is clear and an improvement. However, I've leaned towards generics being a loss for golang. Only use it when you can't solve a problem with a more verbose option. This would lean towards a config struct with a nil guard in new. As always, happy to discuss further